gnoi/os/manager.go

```diff
@@ -0,0 +1,99 @@
+/* Copyright 2020 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package os
+
+import (
+	"bytes"
+	"fmt"
+	"time"
+)
+
+// OS represents the data for a singular operating system.
+type OS struct {
+	Name,
+	Hash,
+	Version string
+	BuildDate time.Time
+	Data      *bytes.Buffer
+}
+
+// Manager for storing data on OS's.
```

Comment:
I like how you have set this up has if we are running real OSs in the Target =)

I think we can simplify here a bit (this is what I think, I may be wrong, please point out if so):
1) Manager only really needs: runningOS, factoryOS and []installeOSs string. We dont do anything with all the other data.
2) The hash validation is only done at the reception of the binary data, so we dont need to store the hashes;
3) Manager should support all versions, we can signal an unsupported version in the OS file directly by using a cookie, example:

OS file:
hash (for everything after this line)\
cookie (just a unique id string) \
version (string) \
random padding \
<EOF>

Reply:
So for now just keep OS here and later move it?